hack/apidiff.sh

```diff
@@ -82,24 +82,17 @@ shift $((OPTIND - 1))
 # Check specific directory or everything.
 targets=("$@")
 if [ ${#targets[@]} -eq 0 ]; then
-    # This lists all entries in the go.work file as absolute directory paths.
-    kube::util::read-array targets < <(go list -f '{{.Dir}}' -m)
+    shopt -s globstar
+    # Modules are discovered by looking for go.mod rather than asking go
+    # to ensure that modules that aren't part of the workspace and/or are
+    # not dependencies are checked too.
+    # . and staging are listed explicitly here to avoid _output
+    for module in ./go.mod ./staging/**/go.mod; do
+        module="${module%/go.mod}"
+        targets+=("$module")
+    done
 fi
 
-# Sanitize paths:
-# - We need relative paths because we will invoke apidiff in
-#   different work trees.
-# - Must start with a dot.
```

Comment:
Removing this code breaks manual invocations like `apidiff.sh k8s.io/client-go`.

Reply:
let me bring it back, as it does not hurt

Reply:
hmm, `apidiff.sh k8s.io/client-go` didn't work before.

```
./hack/apidiff.sh -t v1.32.0-rc.0 -r v1.32.0-beta.0 k8s.io/client-go
Checking d0627a38043fc62a96e6f21b940c2ee714ef97c8 (= v1.32.0-rc.0) for API changes since 0c41a48301eb3619da6460dfdf2db6eb92560807 (= v1.32.0-beta.0).
Installing apidiff into /tmp/tmp.VkRa4rQCK2.
Preparing worktree (detached HEAD 8046362e6ff)
Updating files: 100% (26242/26242), done.
HEAD is now at 8046362e6ff Release commit for Kubernetes v1.32.0-rc.0
loading ./k8s.io/client-go: -: pattern ./k8s.io/client-go/...: lstat ./k8s.io/client-go/: no such file or directory
!!! [1130 06:08:05] Call tree:
!!! [1130 06:08:05]  1: ./hack/apidiff.sh:184 run(...)
!!! [1130 06:08:05]  2: ./hack/apidiff.sh:194 inWorktree(...)
!!! [1130 06:08:05]  3: ./hack/apidiff.sh:199 inTarget(...)
!!! [1130 06:08:05] Call tree:
!!! [1130 06:08:05]  1: ./hack/apidiff.sh:194 inWorktree(...)
!!! [1130 06:08:05]  2: ./hack/apidiff.sh:199 inTarget(...)

```

This was needed because the `targets` had the absolute path
```
    # This lists all entries in the go.work file as absolute directory paths.
    kube::util::read-array targets < <(go list -f '{{.Dir}}' -m)
```

but now we obtain the relative path directly so no need to sanitize.

I don't feel we need to support passing the module directly for this tool, but in any case that will be better a separate PR

Reply:
Ack. I thought it had worked, but I was mistaken.

/lgtm
/approve